system/file.go

```diff
@@ -214,3 +216,23 @@ func realPath(path string) (string, error) {
 
 	return realPath, err
 }
+
+func (f *DefFile) Md5() (string, error) {
```

Comment:
The `result` variable can be dropped altogether, this code should do the same as the orig:
```golang
func (f *DefFile) Md5() (string, error) {
        if err := f.setup(); err != nil {
                return "", err
        }

        fh, err := os.Open(f.realPath)
        if err != nil {
                return "", err
        }
        defer fh.Close()

        hash := md5.New()
        if _, err := io.Copy(hash, fh); err != nil {
                return "", err
        }

        return fmt.Sprintf("%x", hash.Sum(nil)), nil
}
```